### PR TITLE
Sgt back end

### PR DESCRIPTION
**Pull request by brian-luong**
DELETE
GET
POST
PUT

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:37]: 👍

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    res.status(400).json({
+      error: 'name is required'
+    });
+    return;
+  } else if (!Number.isInteger(score) || score < 0) {
+    res.status(400).json({
+      error: 'score must be a positive integer'
+    });
+    return;
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: A `score` over `100` should also result in a `400 Bad Request` response:

---

image

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        });
+      } else {
+        res.sendStatus(204);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:178]: 👍

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 2 above)
Comment (brian-luong) [reply sgt-back-end/index.js]: @TimelikeClosure updated the range to be 0 to 100

File: sgt-back-end/index.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      res.status(201).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:83]: 👍

File: sgt-back-end/index.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        });
+      } else {
+        res.status(200).json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:145]: 👍

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**